File: csharp/PathTraversal.cs
using System;
using System.IO;

namespace SastSamples
{
    public static class PathTraversal
    {
        public static void Main()
        {
            Console.Write("Enter filename: ");
            var path = Console.ReadLine();

            // CWE-22: no path validation allows directory traversal.
            Console.WriteLine(File.ReadAllText(path));
        }
    }
}

File: go/path_traversal.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

const baseDir = "/var/app/files"

func main() {
	var filename string
	fmt.Print("Enter filename: ")
	fmt.Scan(&filename)

	// CWE-22: filepath.Join without Clean/validation allows directory traversal.
	fullPath := filepath.Join(baseDir, filepath.Base(filename))
	fullPath = filepath.Clean(fullPath)
	if !strings.HasPrefix(fullPath, baseDir) {
		fmt.Println("Error: invalid file path")
		return
	}
	data, err := os.ReadFile(fullPath)
	if err != nil {
		fmt.Printf("Error reading file: %v\n", err)
		return
	}
	fmt.Println(string(data))
}
